Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -99,6 +101,53 @@ public struct TracingDependencyResolverDelegate: DependencyResolverDelegate {
     }
 }
 
+public struct ObservabilityDependencyResolverDelegate: DependencyResolverDelegate {
+    private let observabilityScope: ObservabilityScope
+
+    public init (observabilityScope: ObservabilityScope) {
+        self.observabilityScope = observabilityScope.makeChildScope(description: "DependencyResolver")
+    }
+
+    public func willResolve(term: Term) {
+        self.debug("resolving '\(term.node.package.identity)'")
+    }
+
+    public func didResolve(term: Term, version: Version, duration: DispatchTimeInterval) {
+        self.debug("resolved '\(term.node.package.identity)' @ '\(version)'")
```

Comment:
Is this final, or is there any case in which the resolver might go back and change the version later on?

Reply:
I think it is final, but not 100% to be honest. in any case this PR does not change the resolver semantics, it only routes the information to the observability scope and logs it there in a similar fashion to how it did for the bespoke file based logging

Reply:
Right, understood.  I was thinking more along the lines of whether an IDE could use this callback to, for example, mark packages has having been resolved to a version even while the package resolution proceeds with other packages.  I don't see a problem with this PR per se, other than potential for confusion if there are multiple lines of output suggesting that the package is being resolved to different versions.  Even that case seems pretty understandable to me in a CLI sense though, since a later line of output clearly supersedes an earlier one.  Thanks!

Reply:
> Right, understood. I was thinking more along the lines of whether an IDE could use this callback to, for example, mark a dependency has having been resolved to a version even while the dependency resolution proceeds to resolve other dependencies. I don't see a problem with this PR per se, other than potential for confusion if there are multiple lines of output suggesting that the package is being resolved to different versions. Even that case seems pretty understandable to me in a CLI sense though, since a later line of output clearly supersedes an earlier one. Thanks!

iirc we already have callbacks for making the package is resolved, which we use in the CLI

Reply:
Right, I see `didComputeVersion()` on WorkspaceDelegate — which I think was added earlier this year — and I know that `didResolve()` calls through to that one, and that seems very useful!  But seeing this here raised the question of how "definitive" this is, i.e. whether it is only ever called once.  From the PubGrub code it looks as though it might be definitive, but figured I'd ask.  In any case, just trying to understand the semantics of this one and not a comment on you change in this PR.  Thanks!